tests/compat.sh

```diff
@@ -603,6 +613,11 @@ setup_arguments()
             ;;
     esac
 
+    case $($OPENSSL ciphers ALL) in
```

Comment:
As @gilles-peskine-arm pointed in the [comment](https://github.com/Mbed-TLS/mbedtls/pull/7137#discussion_r1113641823), we need to  consider `ECDH_RSA` cipher suites as well. Should we check `ECDH-ECDSA` and `ECDH-RSA` separately?
I have no idea if there is a scenario which OpenSSL only supports one type of `ECDH-` cipher suites. See that if we introduce `ECDH-RSA` cipher suites for OpenSSL in `compat.sh`, but OpenSSL only supports `ECDH-ECDSA`. Then we're supposed to skip `ECDH-RSA` but actually we don't.
Anyway the check between line 542 and line 617 isn't proper.

Reply:
The code is ok: OpenSSL stopped supporting ECDH-ECDSA cipher suites at the same time as ECDH-RSA. But checking `*ECDH-*` here would be a bit cleaner.

Reply:
`*ECDH-*` should not be used here, as it will match "AECDH-xxx" ciphersuites in a openssl naming flavor. Maybe I could use something like "*ECDH-(ECDSA|RSA)*"